src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -319,14 +322,49 @@ private String generateAssertionsEntryPointClassContent(final Set<ClassDescripti
         ? determineBestEntryPointsAssertionsClassPackage(classDescriptionSet)
         : entryPointClassPackage;
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, PACKAGE, classPackage);
-
+    Set<String> imports;
+    if (entryPointAssertionsClassContent.contains(IMPORTS)) {
+      imports = extractImports(classDescriptionSet);
+      entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, IMPORTS, listNeededImports(imports));
+    } else {
+      imports = new HashSet<>();
+    }
     String allEntryPointsAssertionContent = generateAssertionEntryPointMethodsFor(classDescriptionSet,
-                                                                                  entryPointAssertionMethodTemplate);
+                                                                                  entryPointAssertionMethodTemplate,
+                                                                                  imports);
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, ALL_ASSERTIONS_ENTRY_POINTS,
                                                allEntryPointsAssertionContent);
     return entryPointAssertionsClassContent;
   }
 
+  private static String listNeededImports(Collection<String> imports) {
```

Comment:
I was thinking to have the code looking like:
```java
String imports = resolveImports(classDescriptionSet);
entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, IMPORTS, listNeededImports(imports));
```
and whatever we need to do in `resolveImports` to get the job done.